src/attributes/diagnostics.md

```diff
@@ -193,21 +190,21 @@ when the call expression is a function from an implementation of the trait.
 
 ```rust
 trait Trait {
-  #[must_use]
-  fn use_me(&self) -> i32;
+    #[must_use]
+    fn use_me(&self) -> i32;
 }
 
 impl Trait for i32 {
-  fn use_me(&self) -> i32 { 0i32 }
+    fn use_me(&self) -> i32 { 0i32 }
 }
 
 fn main() {
-  // Violates the `unused_must_use` lint.
-  5i32.use_me();
+    // Violates the `unused_must_use` lint.
+    5i32.use_me();
 }
 ```
 
-When used on a function in an implementation, the attribute does nothing.
+When used on a function in a trait implementation, the attribute does nothing.
```

Comment:
(The below comment is not relevant for this PR)

@varkor @oli-obk We could reasonably make this do something by having the `#[must_use]` trigger when the concrete type is known? E.g.

```rust
trait A { fn b() -> usize; }

struct C;

impl A for C {
    #[must_use]
    fn b() -> usize { 0 }
}

fn main() {
    C::b();
//  ^--- We know statically that `C::b` references the function
//       on `impl A for C { ... }` and so we may trigger the lint...?
}
```